### PR TITLE
Refactor cdn locks tests

### PR DESCRIPTION
**Pull request by ericholguin**
This PR refactors [cdn_locks_test.go](https://github.com/apache/trafficcontrol/blob/master/traffic_ops/testing/api/v4/cdn_locks_test.go) 

Following the new test structure format that uses table driven tests, sub tests, GoLang's test runner, and also leveraging the existing assert functionality and expectation functions.

With these changes tests will be more granular and transparent on what is being tested. Ideally reducing the need to dig through test code to confirm what was tested. In addition this new test structure should also make it easier for new tests to be added as well as reduce redundant code.

**NOTE:** The CDN Lock test that existed in `cachegroups_test.go` has been moved to `cdn_locks_test.go` as it made more sense, due to using the same pre-requisites and functionality.

New output when using `-v`:

```go
--- PASS: TestCDNLocks (4.96s)
    --- PASS: TestCDNLocks/SERVERS_QUEUE_UPDATES (0.15s)
        --- PASS: TestCDNLocks/SERVERS_QUEUE_UPDATES/OK_when_USER_OWNS_LOCK (0.01s)
        --- PASS: TestCDNLocks/SERVERS_QUEUE_UPDATES/FORBIDDEN_when_ADMIN_USER_DOESNT_OWN_LOCK (0.14s)
    --- PASS: TestCDNLocks/TOPOLOGY_QUEUE_UPDATES (0.16s)
        --- PASS: TestCDNLocks/TOPOLOGY_QUEUE_UPDATES/OK_when_USER_OWNS_LOCK (0.00s)
        --- PASS: TestCDNLocks/TOPOLOGY_QUEUE_UPDATES/FORBIDDEN_when_ADMIN_USER_DOESNT_OWN_LOCK (0.15s)
    --- PASS: TestCDNLocks/CACHE_GROUP_UPDATE (0.17s)
        --- PASS: TestCDNLocks/CACHE_GROUP_UPDATE/OK_when_USER_OWNS_LOCK (0.01s)
        --- PASS: TestCDNLocks/CACHE_GROUP_UPDATE/FORBIDDEN_when_ADMIN_USER_DOESNT_OWN_LOCK (0.16s)
    --- PASS: TestCDNLocks/GET (0.00s)
        --- PASS: TestCDNLocks/GET/OK_when_VALID_request (0.00s)
    --- PASS: TestCDNLocks/POST (0.00s)
        --- PASS: TestCDNLocks/POST/CREATED_when_VALID_request (0.00s)
    --- PASS: TestCDNLocks/DELETE (0.14s)
        --- PASS: TestCDNLocks/DELETE/FORBIDDEN_when_NON-ADMIN_USER_DOESNT_OWN_LOCK (0.14s)
        --- PASS: TestCDNLocks/DELETE/OK_when_ADMIN_USER_DOESNT_OWN_LOCK (0.00s)
        --- PASS: TestCDNLocks/DELETE/OK_when_VALID_request (0.00s)
    --- PASS: TestCDNLocks/SNAPSHOT (0.15s)
        --- PASS: TestCDNLocks/SNAPSHOT/OK_when_USER_OWNS_LOCK (0.01s)
        --- PASS: TestCDNLocks/SNAPSHOT/FORBIDDEN_when_ADMIN_USER_DOESNT_OWN_LOCK (0.14s)
```

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Ops Testing

## What is the best way to verify this PR?

Run TO Integration tests

## PR submission checklist
- [x] This PR has tests 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: traffic_ops/testing/api/v4/cdn_locks_test.go [hunk 1]
```diff
@@ -16,470 +16,296 @@ package v4
...
+							err = json.Unmarshal(dat, &cdnLock)
+							assert.NoError(t, err, "Error occurred when unmarshalling request body: %v", err)
+						}
+					}
+
+					switch method {
+					case "GET":
+						t.Run(name, func(t *testing.T) {
+							resp, reqInf, err := testCase.clientSession.GetCDNLocks(testCase.requestOpts)
+							for _, check := range testCase.expectations {
```
Comment (rimashah25) [inline traffic_ops/testing/api/v4/cdn_locks_test.go:180]: In the[ old code base](https://github.com/apache/trafficcontrol/blob/master/traffic_ops/testing/api/v4/cdn_locks_test.go#L141), we were validating fields after we called CDNLocks. Should we follow the same and call validateResponseFields() just like we do for POST case?

**Comment by rimashah25** [comment]
The code changes have taken into account all the test scenarios for current cdn_locks_test. 
The tests pass on local as well.

**Review by zrhoffman — CHANGES_REQUESTED** [review]
Can we use the new `utils.V4TestCase` from #6621 for `methodTests` instead of the anonymous structs?

**Comment by rimashah25** [comment]
> Can we use the new `utils.V4TestCase` from #6621 for `methodTests` instead of the anonymous structs?

Makes sense since we now have that struct in utils.go and can be used for all API test changes.

**Review by zrhoffman — APPROVED** [review]

**Pull request merged**